mpz-core/src/lpn.rs

```diff
@@ -0,0 +1,156 @@
+//! Implement LPN with local linear code.
+//! More especifically, a local linear code is a random boolean matrix with at most D non-zero values in each row.
+
+use crate::{prp::Prp, Block};
+use rayon::prelude::*;
+/// A struct related to LPN.
+/// The `seed` defines a sparse binary matrix `A` with at most `D` non-zero values in each row.
+/// `A` - is a binary matrix with `k` columns and `n` rows. The concrete number of `n` is determined by the input length. `A` will be generated on-the-fly.
+/// `x` - is a `F_{2^128}` vector with length `k`.
+/// `e` - is a `F_{2^128}` vector with length `n`.
+/// Given a vector `x` and `e`, compute `y = Ax + e`.
+/// Note that in the standard LPN problem, `x` is a binary vector, `e` is a sparse binary vector. The way we difined here is a more generic way in term of computing `y`.
+pub struct Lpn<const D: usize> {
+    // The seed to generate the random sparse matrix A.
+    seed: Block,
+
+    // The length of the secret, i.e., x.
+    k: u32,
+
+    // A mask to optimize reduction operation.
+    mask: u32,
```

Comment:
We can use doc comments for these and they will be shown with `cargo doc --document-private-items`

```suggestion
    /// The seed to generate the random sparse matrix A.
    seed: Block,

    /// The length of the secret, i.e., x.
    k: u32,

    /// A mask to optimize reduction operation.
    mask: u32,
```